src/Microsoft.Health.Fhir.Azure/AnonymizationConfigurationArtifactProvider.cs

```diff
@@ -0,0 +1,269 @@
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using EnsureThat;
+using Microsoft.Azure.Storage;
+using Microsoft.Azure.Storage.Blob;
+using Microsoft.Extensions.Caching.Memory;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Microsoft.Health.Fhir.Core.Configs;
+using Microsoft.Health.Fhir.Core.Features.Operations;
+using Microsoft.Health.Fhir.Core.Features.Operations.ConvertData;
+using Microsoft.Health.Fhir.Core.Features.Operations.ConvertData.Models;
+using Microsoft.Health.Fhir.Core.Features.Operations.Export.ExportDestinationClient;
+using Microsoft.Health.Fhir.Core.Features.Operations.Export.Models;
+using Microsoft.Health.Fhir.TemplateManagement.ArtifactProviders;
+using Microsoft.Health.Fhir.TemplateManagement.Client;
+using Microsoft.Health.Fhir.TemplateManagement.Exceptions;
+using Microsoft.Health.Fhir.TemplateManagement.Models;
+using Microsoft.Health.Fhir.TemplateManagement.Utilities;
+using Newtonsoft.Json;
+
+namespace Microsoft.Health.Fhir.Azure
+{
+    public class AnonymizationConfigurationArtifactProvider : IArtifactProvider, IDisposable
+    {
+        private bool _disposed = false;
+        private const string AnonymizationContainer = "anonymization";
+
+        private readonly MemoryCache _cache;
+        private readonly ILogger<AnonymizationConfigurationArtifactProvider> _logger;
+        private IOciArtifactProvider _anonymizationConfigurationCollectionProvider;
+        private IExportClientInitializer<CloudBlobClient> _exportClientInitializer;
+        private IContainerRegistryTokenProvider _containerRegistryTokenProvider;
+
+        private ExportJobConfiguration _exportJobConfiguration;
+        private CloudBlobClient _blobClient;
+
+        public AnonymizationConfigurationArtifactProvider(
+            IExportClientInitializer<CloudBlobClient> exportClientInitializer,
+            IContainerRegistryTokenProvider containerRegistryTokenProvider,
+            IOptions<ExportJobConfiguration> exportJobConfiguration,
+            ILogger<AnonymizationConfigurationArtifactProvider> logger)
+        {
+            EnsureArg.IsNotNull(containerRegistryTokenProvider, nameof(containerRegistryTokenProvider));
+            EnsureArg.IsNotNull(exportClientInitializer, nameof(exportClientInitializer));
+            EnsureArg.IsNotNull(exportJobConfiguration?.Value, nameof(exportJobConfiguration));
+
+            _containerRegistryTokenProvider = containerRegistryTokenProvider;
+            _exportClientInitializer = exportClientInitializer;
+            _exportJobConfiguration = exportJobConfiguration.Value;
+            _logger = logger;
+
+            _cache = new MemoryCache(new MemoryCacheOptions
+            {
+                SizeLimit = _exportJobConfiguration.CacheSizeLimit,
+            });
+        }
+
+        public async Task FetchAsync(string location, Stream targetStream, CancellationToken cancellationToken)
```

Comment:
Can the location string be changed to be an ExportJobRecord object? That would save us having to serialize and deserialize it several times.

Reply:
The purpose of this implementation is to be consistent with the defined interface `IArtifactProvider`. I think type `string` woule be more general.
```
public interface IArtifactProvider
{
    /// <summary>
    /// Fetch artifect used by FHIR server.
    /// </summary>
    /// <param name="location">The location of the artifect. The location string format depends on the artifact provider.</param>
    /// <param name="targetStream">The stream for target artifact content.</param>
    /// <param name="cancellationToken">The cancellation token.</param>
    Task FetchAsync(string location, Stream targetStream, CancellationToken cancellationToken);
}
```

Reply:
Yes, make sense if the method won't work anymore unless the string is an ExportJobRecord. 
I've changed the interfaces.